enketo/openrosa-xpath-evaluator
# ORXE2 Updates for review

Changes included:

* add support for predicates
* drop support for `namespace::`
* run eslint on `test/` directory
* update README to reflect current expression support
* add unit tests; separate integration tests into a separate dir
* remove second implementation for adding/removing custom functions
* drop support for html-encoded greater-than and less-than operators


Still TODO:

* clearly separate Open Rosa-specific vs XPath 1.0 extensions, code & datatypes
* implement missing xpath functions which currently pass through to the wrapped evaluator (see `src/utils/native.js` for some hints); remove `callNative()`
* decide whether we commit to ES5 or ES6 for the main code (tests in ES6+ seems fine - test code has a more controlled runtime environment, and different lint rules)
* remove `namespace::`-related tests completely instead of skipping them?
* redefine `XPathResult.TYPE_*` constants to allow inlining
